File: elearning_web/lib/models/chapter.dart
class Chapter {
  final int? id;
  final String title;
  final String? description;
  final int? orderNumber;
  final String? videoUrl;
  final String? pdfUrl;

  Chapter({this.id, required this.title, this.description, this.orderNumber, this.videoUrl, this.pdfUrl});

  factory Chapter.fromJson(Map<String, dynamic> json) => Chapter(
        id: json['id'] is int ? json['id'] : (json['id'] != null ? int.parse(json['id'].toString()) : null),
        title: json['title'] ?? '',
        description: json['description'],
        orderNumber: json['orderNumber'] is int ? json['orderNumber'] : (json['orderNumber'] != null ? int.parse(json['orderNumber'].toString()) : null),
        videoUrl: json['videoUrl'],
        pdfUrl: json['pdfUrl'],
      );

  Map<String, dynamic> toJson() => {
        if (id != null) 'id': id,
        'title': title,
        if (description != null) 'description': description,
        if (orderNumber != null) 'orderNumber': orderNumber,
        if (videoUrl != null) 'videoUrl': videoUrl,
        if (pdfUrl != null) 'pdfUrl': pdfUrl,
      };
}

File: src/main/java/com/example/elearning/entities/Chapter.java
package com.example.elearning.entities;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.*;
import com.fasterxml.jackson.annotation.JsonFormat;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "chapter",
        uniqueConstraints = {@UniqueConstraint(columnNames = {"course_id", "order_number"})})
public class Chapter {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @ManyToOne
    @JoinColumn(name = "course_id", nullable = false)
    @JsonBackReference
    private Course course;


    @Column(name = "order_number", nullable = false)
    private Integer orderNumber;

    @Column(name = "video_url", columnDefinition = "TEXT")
    private String videoUrl;

    @Column(name = "pdf_url", columnDefinition = "TEXT")
    private String pdfUrl;
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
    }
//    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
//    private LocalDateTime createdAt;
}
